lib/k6-summary/0.0.2/index.js

```diff
@@ -0,0 +1,501 @@
+    decorate = function (text, color /*, ...rest*/) {
+      var result = '\x1b[' + color
+      for (var i = 2; i < arguments.length; i++) {
+        result += ';' + arguments[i]
+      }
+      return result + 'm' + text + '\x1b[0m'
+    }
+  }
+
+  Array.prototype.push.apply(
+    lines,
+    summarizeGroup(mergedOpts.indent + '    ', data.root_group, decorate)
+  )
+
+  Array.prototype.push.apply(lines, summarizeMetrics(mergedOpts, data, decorate))
+
+  return lines.join('\n')
+}
+
+var replacements = {
+  '&': '&amp;',
+  '<': '&lt;',
+  '>': '&gt;',
+  "'": '&#39;',
+  '"': '&quot;',
+}
+
+function escapeHTML(str) {
+  // TODO: something more robust?
+  return str.replace(/[&<>'"]/g, function (char) {
+    return replacements[char]
+  })
+}
+
+function generateJUnitXML(data, options) {
+  var failures = 0
+  var cases = []
+
+  forEach(data.metrics, function (metricName, metric) {
+    if (!metric.thresholds) {
+      return
+    }
+    forEach(metric.thresholds, function (thresholdName, threshold) {
+      if (threshold.ok) {
+        cases.push(
+          '<testcase name="' + escapeHTML(metricName) + ' - ' + escapeHTML(thresholdName) + '" />'
+        )
+      } else {
+        failures++
+        cases.push(
+          '<testcase name="' +
+            escapeHTML(metricName) +
+            ' - ' +
+            escapeHTML(thresholdName) +
+            '"><failure message="failed" /></testcase>'
+        )
+      }
+    })
+  })
+
+  var name = options && options.name ? escapeHTML(options.name) : 'k6 thresholds'
+
+  return (
+    '<?xml version="1.0"?>\n<testsuites tests="' +
+    cases.length +
+    '" failures="' +
+    failures +
+    '">\n' +
+    '<testsuite name="' +
+    name +
+    '" tests="' +
+    cases.length +
+    '" failures="' +
+    failures +
+    '">' +
+    cases.join('\n') +
+    '\n</testsuite >\n</testsuites >'
+  )
+}
+
+/**
+ * Generate a junit xml string from the summary of a k6 run considering each checks as a test case
+ * @param {*} data
+ * @param {String} suiteName Name of the test ex., filename
+ * @returns junit xml string
+ */
+function generateJUnitXMLForChecks(data, suiteName) {
+  var failures = 0
+  var cases = []
+  var time = data.metrics.iteration_duration.values.max
+    ? data.metrics.iteration_duration.values.max
+    : 0
+  suiteName = suiteName ? escapeHTML(suiteName) : 'k6 checks'
+  if (data.root_group.hasOwnProperty('checks') && data.root_group.checks.length > 0) {
+    var checks = data.root_group.checks
+    checks.forEach((check) => {
+      if (check.passes === 1) {
+        cases.push(`<testcase name="${escapeHTML(check.name)}"/>`)
+      } else {
+        failures++
+        cases.push(
+          `<testcase name="${escapeHTML(check.name)}"><failure message="failed"/></testcase>`
+        )
+      }
+    })
+  }
+  return (
+    `<?xml version="1.0" encoding="UTF-8" ?>\n<testsuites tests="${cases.length}" ` +
+    `failures="${failures}\" time="${time}">\n` +
```

Comment:
Hmm this seems wrong. According to https://llg.cubic.org/docs/junit/, this should be
> Time taken (in seconds) to execute the tests in the suite 

We don't have that data readily available in `handleSummary()` yet, though we could easily add it (e.g. in https://github.com/k6io/k6/pull/1975).

Reply:
i did not find anything closer than `iteration_duration` in the `summary.json` to find the time taken to execute the tests

Reply:
Yeah... :disappointed: We can correct that in k6 v0.33.0, since I imagine it will be useful for other things as well. I added a note about it in https://github.com/k6io/k6/pull/1975#issuecomment-823051968